### PR TITLE
Lazy load repositories

### PR DESCRIPTION
**Pull request by joglr**
This PR implements lazy loading of repositories on the main page, such that the application loads faster and then the data is gracefully loaded into view using [`defer`](https://remix.run/docs/en/main/utils/defer) in remix.

### CODE REVIEW

**Comment by joglr** [comment]
This can also be extended to the visualization UI, such that the application shows up faster with basic information, whereas the analyzation can be streamed as it is available. Did you @dawidwoz experiment with this previously?

File: src/analyzer/model.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export interface HydratedGitBlobObject extends GitBlobObject {
   lastChangeEpoch?: number
   isBinary?: boolean
   unionedAuthors?: Record<AuthorshipType, Record<string, number>>
-  dominantAuthor?: Map<AuthorshipType, [string, number]>
+  dominantAuthor?: Record<AuthorshipType, [string, number]>
```
Comment (dawidwoz) [inline src/analyzer/model.ts:78]: As `Record` does have not order options we can reply on -[Discussion on StackOverflow]( https://stackoverflow.com/questions/70781123/how-to-prevent-typescript-recordstring-any-from-sorting-object-properties-to). 

Would we not need to convert it to Map or array at some point? Is this change needed to make lazy load repositories work?

**Comment by dawidwoz** [comment]
> This can also be extended to the visualization UI, such that the application shows up faster with basic information, whereas the analyzation can be streamed as it is available. Did you @dawidwoz experiment with this previously?

I have tried this but I gave up at some point. The main challenge was to define the desired behaviour so it can be logical and intuitive for a user. For example, if a user goes to the "top contributor" and we don't have all data, what should we do? There are many options for what to do, how to split the tasks, and what information a user should get. I believe it is a good technical and UX challenge for a long discussion. 

This PR goes in the right direction. I left a tiny comment but generally, I think it is a valuable change. 

Good job! 🙂

**Review by dawidwoz — APPROVED** [review]

File: src/analyzer/model.ts (hunk 1 above)
Comment (joglr) [reply src/analyzer/model.ts:78]: This was necessary when moving away from typedjson, which enabled using remix `defer` since we no longer revive the JSON and can only use JSON primitives. The ordering of this does not matter, since it was only used for switching between git blame and all history (which is currently not enabled), so I think this change is fine.

Comment (dawidwoz) [reply src/analyzer/model.ts:78]: I don't have rights to resolve the conversation but you answered my concern, so it should be resolved.
